dll/win32/shell32/folders/CDrivesFolder.cpp

```diff
@@ -295,29 +295,31 @@ HRESULT CALLBACK DrivesContextMenuCallback(IShellFolder *psf,
         QCMINFO *pqcminfo = (QCMINFO *)lParam;
 
         UINT idCmdFirst = pqcminfo->idCmdFirst;
+        UINT idCmd = 0;
         if (!(dwFlags & FILE_READ_ONLY_VOLUME) && nDriveType != DRIVE_REMOTE)
         {
             /* add separator and Format */
-            UINT idCmd = idCmdFirst + CMDID_FORMAT;
+            idCmd = idCmdFirst + CMDID_FORMAT;
             _InsertMenuItemW(pqcminfo->hmenu, pqcminfo->indexMenu++, TRUE, 0, MFT_SEPARATOR, NULL, 0);
             _InsertMenuItemW(pqcminfo->hmenu, pqcminfo->indexMenu++, TRUE, idCmd, MFT_STRING, MAKEINTRESOURCEW(IDS_FORMATDRIVE), MFS_ENABLED);
         }
         if (nDriveType == DRIVE_REMOVABLE || nDriveType == DRIVE_CDROM)
         {
             /* add separator and Eject */
-            UINT idCmd = idCmdFirst + CMDID_EJECT;
+            idCmd = idCmdFirst + CMDID_EJECT;
             _InsertMenuItemW(pqcminfo->hmenu, pqcminfo->indexMenu++, TRUE, 0, MFT_SEPARATOR, NULL, 0);
             _InsertMenuItemW(pqcminfo->hmenu, pqcminfo->indexMenu++, TRUE, idCmd, MFT_STRING, MAKEINTRESOURCEW(IDS_EJECT), MFS_ENABLED);
         }
         if (nDriveType == DRIVE_REMOTE)
         {
             /* add separator and Disconnect */
-            UINT idCmd = idCmdFirst + CMDID_DISCONNECT;
+            idCmd = idCmdFirst + CMDID_DISCONNECT;
             _InsertMenuItemW(pqcminfo->hmenu, pqcminfo->indexMenu++, TRUE, 0, MFT_SEPARATOR, NULL, 0);
             _InsertMenuItemW(pqcminfo->hmenu, pqcminfo->indexMenu++, TRUE, idCmd, MFT_STRING, MAKEINTRESOURCEW(IDS_DISCONNECT), MFS_ENABLED);
         }
 
-        pqcminfo->idCmdFirst += 3;
+        if (idCmd)
+            pqcminfo->idCmdFirst = (idCmd + 2);
```

Comment:
If you update the CMDID's to be 0-based, then  you can do +1 here,
which makes more sense since this is supposed to point to the next _free_ item.

Reply:
I have applied your suggestion, it seems to break menu item position:
![VirtualBox_ReactOS_08_11_2022_22_48_52](https://user-images.githubusercontent.com/578406/200661099-a07a36d2-3233-4827-8846-c3e0f948359c.png)

Reply:
That this does not break more often is a game of chance,
looks like we do not treat those menu's correctly at all.
We regularly confuse index and id...

Reply:
I guess I'll commit this one in current state, to avoid introducing one more regresso. 
:coffee:  :sweat_smile:  This has to be investigated deeper at some point.

Reply:
This perfectly highlights a difference between ours and windows' menu building system that I knew existed, but did not understand the reason for yet :)

I'll work on that but it might take some time, so this hackfix is good enough for now.

Reply:
I have added `#if 0` with a comment for it. Also mentioned in the commit message. :+1: